### ISSUE TITLE
DataVec: FileSplit (and others) - skip any files with name starting with "."

### ISSUE THREAD

**Issue by AlexDBlack** [eclipse/deeplearning4j#8080]
We should skip by default any files starting with `.`

For example, Apple double format files prefix with `._`
https://en.wikipedia.org/wiki/AppleSingle_and_AppleDouble_formats

For example, attempting to load an `._image.jpg` using ImageRecordReader will obviously fail...

**Comment by saudet** [comment]
It doesn't need to fail, we just need to add support for JPEG 2000 :) 
We probably don't want to read them out anyway.

**Comment by AlexDBlack** [comment]
@saudet sorry, I'm not following what this has to do with JPEG 2000?
If I understand this correctly, `._image.jpg` for example isn't actually a data/image file, it's a Mac OS metadata/resource file that exists in parallel/addition to the "real" `image.jpg` file. That's what we're wanting to filter out here.

OSX also creates `.DS_Store`, and a bunch of other dot files, apparently...

**Comment by saudet** [comment]
Given the error message, it looks like the metadata for images is in JPEG 2000 format:
> Error in pixReadMemJp2k: function not present
> Error in pixReadMem: jp2k: no pix returned

That's apparently the kind of files they were trying to load in issue #2760.

**Comment by agibsonccc** [comment]
Closing — references a removed/rewritten subsystem (DataVec, NLP/Word2Vec, Model Zoo, Spark, UI, or Arbiter). These have been fundamentally rewritten or removed in the upcoming release. Please open a new issue if you experience similar problems with the new version.
